x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/full_time_range_selector/full_time_range_selector_service.ts

```diff
@@ -27,16 +28,27 @@ export async function setFullTimeRange(
   excludeFrozenData?: boolean
 ): Promise<GetTimeFieldRangeResponse> {
   const runtimeMappings = indexPattern.getRuntimeMappings();
+
   const resp = await getTimeFieldRange({
     index: indexPattern.title,
     timeFieldName: indexPattern.timeFieldName,
     query: excludeFrozenData ? addExcludeFrozenToQuery(query) : query,
     ...(isPopulatedObject(runtimeMappings) ? { runtimeMappings } : {}),
   });
-  timefilter.setTime({
-    from: moment(resp.start.epoch).toISOString(),
-    to: moment(resp.end.epoch).toISOString(),
-  });
+
+  if (resp.start.epoch && resp.end.epoch) {
+    timefilter.setTime({
+      from: moment(resp.start.epoch).toISOString(),
+      to: moment(resp.end.epoch).toISOString(),
+    });
+  } else {
+    throw Error(
+      i18n.translate('xpack.dataVisualizer.index.fullTimeRangeSelector.unpopulatedTimeField', {
+        defaultMessage: `No data found for time field {timeFieldName}`,
```

Comment:
>        defaultMessage: `No data found for time field {timeFieldName}`,

Not mandatory, but on first glance it still wasn't clear to me what action I'd need to take. It might be worth adding some concrete advice, for example: "Consider changing the time field for this data view" (though now on further investigation, it doesn't seem like you can do that except via API)

Reply:
Maybe we shouldn't throw an error at all here, but use a separate warning level toast for this case where there is no start / end date returned. This will also get displayed when using the 'Use full data' button for an index which has the time field set correctly, but which contains no documents. The text could then be something like `No results found for data view ${indexPattern.title}`.

Reply:
@lcawl I ended up removing the message for the time field completely because Pete brought up a good one. We wouldn't know if no results found because the time field is incorrect or if the index just have no documents.